src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs

```diff
@@ -111,7 +112,8 @@ private static class AlgorithmActivator
         /// <summary>
         /// Creates a factory that wraps a call to <see cref="Activator.CreateInstance{T}"/>.
         /// </summary>
-        public static Func<T> CreateFactory<T>([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type implementation)
+        [UnconditionalSuppressMessage("AOT", "IL3050", Justification = "MakeGenericType is safe to use because implementation is either a KeyedHashAlgorithm or SymmetricAlgorithm type.")]
+        public static Func<T> CreateFactory<T>([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] Type implementation) where T : class
```

Comment:
Nit: This whole class seems like a really complex way to do what it does - return a Func. I don't know the original reasons, but now with the delegate cachine Roslyn does I would think that a simple lambda would work the same way.

Reply:
I don't know the reason either. It can be looked at in the future. I want to keep this PR focused.

Reply:
Should we open an issue to track looking at this in the future, so it doesn't get forgotten?